packages/application-generic/src/usecases/create-notification-jobs/create-notification-jobs.usecase.ts

```diff
@@ -162,22 +162,30 @@ export class CreateNotificationJobs {
   }
 
   private calculateExpireAt(command: CreateNotificationJobsCommand) {
-    const delayedSteps = command.template.steps.filter(
-      (step) =>
-        step.template?.type === StepTypeEnum.DIGEST ||
-        step.template?.type === StepTypeEnum.DELAY
-    );
-
-    const delay = delayedSteps
-      .map((step) =>
-        this.calculateDelayService.calculateDelay({
-          stepMetadata: step.metadata,
-          payload: command.payload,
-          overrides: command.overrides,
-        })
-      )
-      .reduce((sum, delayAmount) => sum + delayAmount, 0);
+    try {
+      const delayedSteps = command.template.steps.filter(
+        (step) =>
+          step.template?.type === StepTypeEnum.DIGEST ||
+          step.template?.type === StepTypeEnum.DELAY
+      );
 
-    return addMilliseconds(Date.now(), delay);
+      const delay = delayedSteps
+        .map((step) =>
+          this.calculateDelayService.calculateDelay({
+            stepMetadata: step.metadata,
+            payload: command.payload,
+            overrides: command.overrides,
+          })
+        )
+        .reduce((sum, delayAmount) => sum + delayAmount, 0);
+
+      return addMilliseconds(Date.now(), delay);
+    } catch (e) {
+      /*
+       * If the user has entered an incorrect negative delay,
+       * we'll accept it as a temporary solution to enable printing error execution details later in the process when a job is available.
+       */
+      return Date.now();
```

Comment:
won't this be a problem if we expire the entity before it gets executed? like I'm not sure how the MongoDB Atlas will behave in this kind of situation, would it archive it right away?